src/SFML/Graphics/Text.cpp

```diff
@@ -231,6 +294,11 @@ void Text::draw(RenderTarget& target, RenderStates states) const
 
         states.transform *= getTransform();
         states.texture = &m_font->getTexture(m_characterSize);
+
+        // Only draw the outline if there is something to draw
+        if (m_outlineThickness > 0)
```

Comment:
> But if (m_outlineThickness) is used everywhere else in the code to see if the outline is used or not.

Then that's a mistake that must be corrected. We only use "implicit" checks for `bool` and pointers, see [code style](http://www.sfml-dev.org/style.php#type-conversions).